draft-irtf-cfrg-bbs-signatures.md

```diff
@@ -779,8 +787,158 @@ Procedure:
 11. return (scalar_1, ..., scalar_count)
 ```
 
+## Domain Calculation
+
+This operation calculates the domain value, a scalar encoding public parameter information. Optionally, the domain can also include an additional octet string, called the header, meant to encode any additional information the signer may want to "bind" to the signature (like an expiration date, the targeted audience identifier etc.). The domain value will be signed using a specific generator point (`Q_2`) (see [Sign](#sign)) to protect the integrity of those public parameters. The header, if existing, must be known by all parties (i.e., the signer, the prover and verifier), for both the signature and proof to be validated correctly. This means, that the header can be used to enforce certain values (chosen by the signer) to always be disclosed by the prover.
+
+This operation makes use of the `serialize` function, defined in [Section 4.6.1](#serialize).
+
+```
+domain = get_domain(Q_1, Q_2, H_Points, PK, header)
```

Comment:
Calls to this method have PK as the first argument.

Reply:
Maybe `encode_domain` (and `encode_challenge`) would be more descriptive?

Reply:
Or `derive_`, `calculate_`, `hash_`, etc...

Reply:
Done! (used `calculate_`).